federationtypes.go

```diff
@@ -204,6 +219,16 @@ type respStateFields struct {
 	AuthEvents  []Event `json:"auth_chain"`
 }
 
+// XXX: this duplicates RespPeek above and needs to be kept in sync with it
```

Comment:
It looks like you’ve tried to copy a bit of a Go hack — it lets you define a bunch of public fields (so that you can JSON marshal/unmarshal), but because they are in an unexported struct, it prevents things from outside the package manipulating those fields directly.

Normally if you do this, `RespPeek` would just embed the private struct (`type RespPeek struct { respPeekFields }`) but might require some marshalling/unmarshalling wrappers, otherwise you can probably just get rid of `respPeekFields` if it isn’t being used anywhere else deliberately.

Reply:
(In this instance, having internal fields might be overkill so probably best just to remove that, otherwise you also need to define exported functions to get things out of the struct.)